src/update/UpdateManager.cpp

```diff
@@ -62,48 +60,50 @@ VersionPatch UpdateManager::getVersionPatch() const
 
 void UpdateManager::getUpdates()
 {
-    audacity::network_manager::Request request("https://updates.audacityteam.org/feed/latest.xml");
+    const audacity::network_manager::Request request("https://updates.audacityteam.org/feed/latest.xml");
     auto response = audacity::network_manager::NetworkManager::GetInstance().doGet(request);
 
     response->setRequestFinishedCallback([response, this](audacity::network_manager::IResponse*) {
 
+        wxFrame* parent = FindProjectFrame(GetActiveProject());
+        wxASSERT(parent);
+
+        if (!parent) return;
+
         if (response->getError() != audacity::network_manager::NetworkError::NoError)
         {
-            AudacityMessageBox(
-                XO("Unable to connect to Audacity update server."),
+            ShowExceptionDialog(parent,
                 XO("Error checking for update"),
-                wxOK | wxCENTRE,
-                NULL);
+                XO("Unable to connect to Audacity update server."),
+                wxString());
 
             return;
         }
 
         if (!mUpdateDataParser.Parse(response->readAll<VersionPatch::UpdateDataFormat>(), &mVersionPatch))
         {
-            AudacityMessageBox(
-                XO("Update data was corrupted."),
+            ShowExceptionDialog(parent,
                 XO("Error checking for update"),
-                wxOK | wxCENTRE,
-                NULL);
+                XO("Update data was corrupted."),
+                wxString());
 
             return;
         }
 
         if (mVersionPatch.version > CurrentBuildVersion())
         {
-            mParent->CallAfter([this] {
-                UpdatePopupDialog dlg(mParent, this);
+            parent->CallAfter([this, parent] {
```

Comment:
As we also discussed, replace the first `parent` with wxTheApp in case parent is null.

Instead of capturing parent in the lambda, you can repeat `FindProjectFrame(GetActiveProject());` inside it, and that may be safer.  There may be a small change that parent is destroyed between executing CallAfter now, and calling the lambda later, leaving a dangling pointer.